clarify db backup sequence (#642)
digital-asset/docs.daml.com: docs/2.8.1/docs/canton/usermanual/persistence.rst

It is important that the participant's backup is not more recent than that of
the domain's, as that would constitute a ledger fork. Therefore, if you back up
both participant and domain databases sequentially, always back up the participant database before the

> domain. If you are able to perform a complete system backup in a single step (e.g. when using

Reviewer: Shall we clarify here already the domain "internal" order of backups (mediator and domain manager before sequencer)?

Author: You mean bring it up from below?

Reviewer: Not sure what that means 😬.
What I mean is the following: the sentence "_always back up the participant database before the domain_" clarifies the order between participant and domain, but isn't clear that mediator and domain manager databases (in case they are separate nodes) should be backed up before the sequencer database.

Author: Rephrased

It is important that the participant's backup is not more recent than that of
the domain's, as that would constitute a ledger fork. Therefore, if you back up
both participant and domain databases sequentially, the following constraints apply:

- If you run a composite domain node, always back up the participant database before the domain.

- If you run different nodes of the domain separately (sequencer, mediators, domain topology manager):

>   - Back up the mediators, the domain manager and participants before the sequencer;
>     otherwise they may not be able to reconnect to the sequencer (``ForkHappened``).
>     The relative order of mediators, domain topology manager, and participants does not matter.
> 
>   - On a :ref:`permissioned domain <permissioned-domains>`, make sure that
>     no participant is registered at the domain topology manager
>     between the backup of the domain topology manager and the backup of the sequencer.
>     Otherwise the participant may not work correctly after the recovery.

Reviewer 2: ```suggestion
  - Back up the mediators, the domain manager, and participants before the sequencer;
    otherwise, they may not be able to reconnect to the sequencer (``ForkHappened``).
    The relative order of mediators, domain topology manager, and participants does not matter.

  - On a :ref:`permissioned domain <permissioned-domains>`, make sure
    no participant is registered with the domain topology manager
    between the backup of the domain topology manager and the backup of the sequencer.
    Otherwise, the participant may not work correctly after the recovery.
```

If you are able to perform a complete system backup in a single step (e.g. when using
a cloud RDS) you have to ensure that no component is making writes to the db while the backup is ongoing.

> If you are using a domain integration (Fabric, Besu), then backup the sequencer node before backing

Author: These are still used in the 2.x release lines and we owe this explanation to our users (it is quite arcane article, so if a user ends up here, they must already be quite deep in the Fabric or Besu integration)
